lua/navigator.lua

```diff
@@ -229,11 +229,7 @@ M.setup = function(cfg)
   require('navigator.hierarchy')
   require('navigator.implementation')
 
-  if not cfg.lsp then
-    cfg.lsp= { diagnostic = _NgConfigValues.lsp.diagnostic }
-  elseif not cfg.lsp.diagnostic then
-    cfg.lsp.diagnostic = _NgConfigValues.lsp.diagnostic
-  end
+  cfg.lsp = cfg.lsp or _NgConfigValues.lsp
```

Comment:
if `cfg` has other fields except `diagnostic`, it will overwrite the other fields?

Reply:
if 
```
cfg = {lsp={format_on_save=false}}
require('navigator.diagnostics').config(cfg)
```
it is same as
```
require('navigator.diagnostics').config()
```

it will fallback to default setup here
```
function M.config(cfg)
  cfg = cfg
    or {
      underline = true,
      virtual_text = true,
      signs = { _NgConfigValues.icons.diagnostic_err },
      update_in_insert = false,
    }
  vim.diagnostic.config(cfg)
end
```

Reply:
you are correct.
If the subsequent code uses this variable, there may be problems, you need to be careful